persistence/petsPersistence.test.js

```diff
@@ -0,0 +1,217 @@
+                }       
+            });
+        })        
+    })
+
+
+    describe('#insertPet', ()=>{
+        beforeEach(()=>{
+            fileReader = jest.fn().mockResolvedValue(petList);
+            writeFile = jest.fn().mockResolvedValue(petMock);
+            instance = new petsPersistence(fileReader, writeFile);
+        })
+
+        test('Check if the pet is inserted', async ()=>{
+            let pet = new Pet();
+            let result = await instance.insertPet(pet);
+            expect(result).toBeInstanceOf(Pet);
+        })
+
+        describe('When and error occurs', ()=>{
+
+            beforeEach(()=>{
+                fileReader = jest.fn().mockResolvedValue(petList);
+                writeFile = jest.fn().mockRejectedValue(new Error());
+                instance = new petsPersistence(fileReader, writeFile);
+            })
+
+            test('could not read file', async() =>{
+                try {
+                    let pet = new Pet();
+                    await instance.insertPet(pet);
+                } catch (error) {
+                    expect(error).toBeInstanceOf(CannotReadFile);
+                }
+            })
+        })
+    })
+    describe('#updatePet', ()=>{
+
+        beforeEach(()=>{
+            fileReader = jest.fn().mockResolvedValue(petList);
+            writeFile = jest.fn().mockResolvedValue(petMock);
+            instance = new petsPersistence(fileReader, writeFile);
+        })
+
+        test('pet object on the return of the function', async ()=>{
+            let petUpdate = new Pet();
+            petUpdate.id = 2;
+            petUpdate.name = 'teste teste';
+            let petReturn = await instance.updatePet(petUpdate);
+            expect(petReturn).toBeInstanceOf(Pet);
+        })
+
+        describe('When an error occurs', ()=>{
+
+            beforeEach(() => {
+                fileReader = jest.fn().mockResolvedValue(petList);
+                writeFile = jest.fn().mockRejectedValue(new Error());
+                instance = new petsPersistence(fileReader, writeFile);
+            })
+
+            test('could not read the file', async ()=>{
+                try {
+                    let petUpdate = new Pet();
+                    await instance.updatePet(petUpdate);
+                } catch (error) {
+                    expect(error).toBeInstanceOf(NotFoundException);
+                }
+            })
+        })
+    })
+
+    describe('#deletePet', ()=> {
+
+        beforeEach(()=>{
+            fileReader = jest.fn().mockResolvedValue(petList);
+            writeFile = jest.fn().mockResolvedValue(petMock);
+            instance = new petsPersistence(fileReader, writeFile);
+        })
+
+        test('pet Object on the return of the function', async()=>{            
+            let petDeleted = await instance.deletePet(2);
+            expect(petDeleted).toBeInstanceOf(Pet);
+        });
+
+        describe('When an error occurs', ()=>{
```

Comment:
in this case, you can have two types of error `CannotReadFile` and `NotFoundException`. You need to check the behavior for these two.
```
describe("when getFileData raises CannotReadFile error")
describe("when the given Pet is not found")
```

Actually all methods must handle `CannotReadFile` exception.